### PR TITLE
WIP Requirable factories

### LINKED ISSUE #595: Factories aren't require-able independently
The factories should be `require`'able independently. Using them in your own project is not possible at the moment, because each factory has lots of dependencies, that are not met in the factory file itself.

### PR DESCRIPTION
**Pull request by Sinetheta**
This is mostly just a lot of unfortunate tree traversal, as we laboriously trace out a dependency graph.

Resolves #595

### CODE REVIEW

File: core/lib/spree/testing_support/factories/shipping_method_factory.rb [hunk 1]
```diff
@@ -1,6 +1,12 @@
+require 'spree/testing_support/factories/calculator_factory'
+require 'spree/testing_support/factories/shipping_category_factory'
+require 'spree/testing_support/factories/zone_factory'
+
 FactoryGirl.define do
   factory :base_shipping_method, class: Spree::ShippingMethod do
-    zones { |a| [Spree::Zone.global] }
+    zones do
+      [Spree::Zone.find_by(name: 'GlobalZone') || FactoryGirl.create(:global_zone)]
+    end
```
Comment (mtomov) [inline core/lib/spree/testing_support/factories/shipping_method_factory.rb]: :+1:

**Comment by tvdeyen** [comment]
Yeah. What a Christmas present!!! Thank you so very much @Sinetheta 🎄🎄🎅🏻🎅🏻

**Comment by mamhoff** [comment]
@Sinetheta This is really, really nice stuff. There's a bit of stuff missing; if you want help - can I PR against your branch?

**Comment by Sinetheta** [comment]
That sounds great @mamhoff. I meant to finish this over the holidays but got a little side-tracked.

In order to test these I pulled https://github.com/solidusio/solidus/blob/master/core/spec/spec_helper.rb#L36 out of the spec_helper and was using a simple `test_spec.rb` where I would just swap out the factory I wanted to isolate.

```rb
require 'spec_helper'
require 'spree/testing_support/factories/address_factory'

describe 'things', :type => :model do
  let(:address) { create(:address) }

  it "should pass" do
    expect(address).to be_an(Spree::Address)
  end
end
```

The only problem is that I wasn't being super thorough testing *all* factories in a file, or checking all of the traits. I was planning on doing little `--fixup`s where needed.

By all means send PRs to my branch, or just open a second PR with your own fork. If you finish first I can shut this one down, or I can just cherry-pick any progress you make. It will probably be a few days before I'd get a chance to get back to this anyways.

Thanks!

**Comment by cbrunsdon** [comment]
Okay, it looks like all these commits are in #627 so I'm killing this PR

**Pull request closed without merging**